test/lang/test_srt_engine.py

```diff
@@ -0,0 +1,31 @@
+import unittest
+
+import sglang as sgl
+from sglang.test.test_utils import DEFAULT_MODEL_NAME_FOR_TEST
+
+
+class TestSRTBackend(unittest.TestCase):
+    engine = None
+
+    @classmethod
+    def setUpClass(cls):
+        cls.engine = sgl.Engine(model_path=DEFAULT_MODEL_NAME_FOR_TEST)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.engine.shutdown()
+
+    def test_generate(self):
+        prompts = [
+            "The capital of China is",
+            "The square root of 144 is",
+        ]
+
+        outputs = self.engine.generate(prompts, {"max_new_tokens": 128})
+
+        assert "Beijing" in outputs[0]["text"]
+        assert "12" in outputs[1]["text"]
```

Comment:
Can we add a more strict comparison? Like assert a long output is exactly same with using the old interface of Runtime.

Reply:
The consistency check is a bit tricky because:

1. It's not possible to pass a random seed from the client side; it can only be set during engine/runtime initialization.
2. When using the runtime, the client runs a warmup thread, which adds an extra dummy request for "The capital city of France". This causes the random seed to advance.

As a result, both scenarios have different random seeds when processing "what is AGI?".

Engine:
```
GenerateReqInput(text='what is AGI?',
```

Runtime:
```
GenerateReqInput(text='The capital city of France is'
GenerateReqInput(text='what is AGI?',
```

This difference in random seed states leads to potential inconsistencies when handling the same input ("what is AGI?") between the Engine and Runtime implementations.

What if we make the test case harder? or anyway to fix random seed per request?

Reply:
discussed offline, setting `sampling_backend='pytorch'` and temp = 0 can be deterministic regardless of the random seed